accounts/checkers.go

```diff
@@ -0,0 +1,615 @@
+	GetNodeInfoPassThrough = mid.NewPassThrough(
+		&lnrpc.NodeInfoRequest{}, &lnrpc.NodeInfo{},
+	)
+
+	// PendingChannelsEmptyRewriter is a response re-writer that returns a
+	// response to PendingChannels with zero channels shown.
+	PendingChannelsEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.PendingChannelsRequest, *lnrpc.PendingChannelsResponse,
+	]()
+
+	// ListChannelsEmptyRewriter is a response re-writer that returns a
+	// response to ListChannels with zero channels shown.
+	ListChannelsEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.ListChannelsRequest, *lnrpc.ListChannelsResponse,
+	]()
+
+	// ClosedChannelsEmptyRewriter is a response re-writer that returns a
+	// response to ClosedChannels with zero channels shown.
+	ClosedChannelsEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.ClosedChannelsRequest, *lnrpc.ClosedChannelsResponse,
+	]()
+
+	// WalletBalanceEmptyRewriter is a response re-writer that returns a
+	// response to WalletBalance with a zero balance shown.
+	WalletBalanceEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.WalletBalanceRequest, *lnrpc.WalletBalanceResponse,
+	]()
+
+	// GetTransactionsEmptyRewriter is a response re-writer that returns a
+	// response to GetTransactions with zero transactions shown.
+	GetTransactionsEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.GetTransactionsRequest, *lnrpc.TransactionDetails,
+	]()
+
+	// ListPeersEmptyRewriter is a response re-writer that returns a
+	// response to ListPeers with zero peers shown.
+	ListPeersEmptyRewriter = mid.NewResponseEmptier[
+		*lnrpc.ListPeersRequest, *lnrpc.ListPeersResponse,
+	]()
+)
+
+// CheckerMap is a type alias that maps gRPC request URIs to their
+// rpcmiddleware.RoundTripChecker types.
+type CheckerMap map[string]mid.RoundTripChecker
+
+// AccountChecker is a type that can check all account related requests,
+// including invoices, payments and account balances.
+type AccountChecker struct {
+	checkers CheckerMap
+}
+
+// NewAccountChecker creates a new account checker that can keep track of all
+// account related requests, including invoices, payments and account balances.
+func NewAccountChecker(service Service,
+	chainParams *chaincfg.Params) *AccountChecker {
+
+	// sendResponseHandler is a response handler function that is used by
+	// multiple RPC checkers for checking an RPC response sent for a payment
+	// attempt.
+	sendResponseHandler := func(ctx context.Context,
+		r *lnrpc.SendResponse) (proto.Message, error) {
+
+		status := lnrpc.Payment_IN_FLIGHT
+		if len(r.PaymentError) > 0 {
+			status = lnrpc.Payment_FAILED
+		}
+
+		hash, err := lntypes.MakeHash(r.PaymentHash)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing payment hash: %v",
+				err)
+		}
+
+		route := r.PaymentRoute
+		totalAmount := int64(0)
+		if route != nil {
+			totalAmount = route.TotalAmtMsat + route.TotalFeesMsat
+		}
+
+		return checkSendResponse(
+			ctx, service, status, hash, totalAmount,
+		)
+	}
+
+	checkers := CheckerMap{
```

Comment:
Will any calls the aren't part of this map succeed as normal, or error out due to insufficient permissions?